framework/db/QueryRecord.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace yii\db;
+
+use Yii;
+use yii\base\NotSupportedException;
+use yii\base\InvalidConfigException;
+use yii\helpers\Inflector;
+use yii\helpers\StringHelper;
+
+/**
+ * QuerRecord is the base class for classes representing query statement data in terms of objects.
+ *
+ * To declare an QueryRecord class you need to extend [[\yii\db\QueryRecord]] and
+ * implement the `query` method:
+ *
+ * ```php
+ * <?php
+ *
+ * class InvoicePaid extends \yii\db\QueryRecord
+ * {
+ *     public function attributes()
+ *     {
+ *         return ['invoice_id', 'total'];
+ *     }
+ *
+ *     public static function query()
+ *     {
+ *         return (new Query())
+ *             ->select(['invoice_id', 'total' => 'sum(value)'])
+ *             ->from('payment')
+ *             ->groupBy(['invoice_id']);
+ *     }
+ * }
+ * ```
+ *
+ * The `query` method only has to return the [[\yii\db\Query]] representation of the class.
+ *
+ * Below is an example showing some typical usage of QueryRecord:
+ *
+ * ```php
+ * $paids = InvoicePaid::find()
+ *              ->where(['>', 'total', 3426])
+ *              ->all();
+ * ```
+ *
+ * Use as relation
+ * ```php
+ * class Invoive extends ActiveRecord
+ * {
+ *     ...
+ *     public function getPaid()
+ *     {
+ *         return $this->hasOne(InvoicePaid::className(), ['invoice_id' => 'id']);
+ *     }
+ * }
+ * ```
+ *
+ * Then access it as
+ * ```php
+ * $invoices = Invoice::find()
+ *                 ->alias('i')
+ *                 ->joinWith(['paid p'])
+ *                 ->where('[[i.value]] > [[p.total]]')
+ *                 ->all();
+ *
+ * foreach (Invoice::find()->with('paid')->all() as $model) {
+ *     echo $model->paid->total;
+ * }
+ * ```
+ * For more details and usage information on ActiveRecord, see the [guide article on ActiveRecord](guide:db-active-record).
+ *
+ * @method ActiveQuery hasMany($class, array $link) see [[BaseActiveRecord::hasMany()]] for more info
+ * @method ActiveQuery hasOne($class, array $link) see [[BaseActiveRecord::hasOne()]] for more info
+ *
+ * @author Misbahul D Munir <misbahuldmunir@gmail.com>
+ * @since 2.10
+ */
+class QueryRecord extends BaseActiveRecord
```

Comment:
disagree.
This class should can be implemented with minimal code. I mean, just implement `query()` and `attributes()`.